pkg/nubuild/build/build.go

```diff
@@ -82,11 +86,40 @@ type buildStep struct {
 	Func    func() error
 }
 
-func NewBuilder(parentLogger nuclio.Logger, options *Options) *Builder {
+func findFunctionPath(options *Options, path string) (err error) {
+	if strings.HasPrefix(path, "http://") || strings.HasPrefix(path, "https://") {
+		out, err := ioutil.TempFile("", "")
+		if err != nil {
+			return
+		}
+		defer out.Close()
+
+		response, err := http.Get(path)
+		if err != nil {
+			return
+		}
+		defer response.Body.Close()
+
+		_, err = io.Copy(out, response.Body)
+		if err != nil {
+			return
+		}
+		options.FunctionPath = out.Name()
```

Comment:
`Close` the file here

Reply:
Why not use `defer`? I have it right after I create the temp file.